addons/crm_livechat/static/src/core/thread_actions.js

```diff
@@ -0,0 +1,45 @@
+import { LivechatCommandDialog } from "@im_livechat/core/common/livechat_command_dialog";
+
+import { threadActionsRegistry } from "@mail/core/common/thread_actions";
+import "@mail/discuss/call/common/thread_actions";
+
+import { useComponent } from "@odoo/owl";
+
+import { _t } from "@web/core/l10n/translation";
+import { usePopover } from "@web/core/popover/popover_hook";
+
+threadActionsRegistry.add("create-lead", {
+    close: (component, action) => action.popover?.close(),
+    component: LivechatCommandDialog,
+    componentProps: (action) => ({
+        close: () => action.close(),
+        commandName: "lead",
+        placeholderText: _t("e.g. Product pricing"),
+        title: _t("Create Lead"),
+        icon: "fa fa-handshake-o",
+    }),
+    condition: (component) =>
+        component.store.has_access_create_lead && component.thread?.hasLeadAction,
```

Comment:
Instead of relying on `hasLeadAction` on thread, I would suggest to make conditional of thread actions patchable. This would prevent the indirection to the `thread_model` file and keep thread action relevant code in this file (and patches).

There's an ongoing PR to be merged soon in master that does this: https://github.com/odoo/odoo/pull/199648/files#diff-df32f63e089b7be74c220ca5ee20d81a8499d38558fd91b912fd58a32f45d0a3R209

You could copy this code and patch the same way. The 1st PR to be merged with it will win, and solving conflicts is very easy as they use the same technique.